src/features/peers/redux/__test__/middleware.test.ts

```diff
@@ -0,0 +1,50 @@
+import { topPeersMiddleware } from "../middleware";
+import { BOOTSTRAP } from "store";
+import { PeersData } from "../actions";
+
+describe("Testing the Peers Middleware", () => {
+  let mockSocket: any;
+  let socketService: any;
+  let store: any;
+  let next: jest.Mock;
+
+  beforeEach(() => {
+    const payload = [{ symbol: "AAPL", name: "APPLE" }];
+    mockSocket = {
+      on: jest.fn((name: string, callback: (payload: PeersData[]) => void) => {
+        callback(payload);
+      })
+    };
+    socketService = {
+      get: () => mockSocket
+    };
+
+    const peerState = { selectedTopPeers: [{ symbol: "AAPL", name: "APPLE" }] };
+
+    store = {
+      getState: jest.fn(() => peerState),
+      dispatch: jest.fn()
+    };
+
+    next = jest.fn();
+  });
+
+  test.only("socket.on and dispatch been called", () => {
```

Comment:
nit: Might be nicer to split your socket creation and dispatch tests into two tests?

Reply:
I think my dispatch is within the socket, so I did it in one test